hack/dind-cluster.sh

```diff
@@ -255,74 +133,103 @@ cluster's rc file to configure the bash environment:
 }
 
 function stop() {
-  echo "Cleaning up docker-in-docker containers"
+  local config_root=$1
+  local cluster_id=$2
+
+  echo "Stopping dind cluster '${cluster_id}'"
 
   local master_cid="$(${DOCKER_CMD} ps -qa --filter "name=${MASTER_NAME}")"
   if [[ "${master_cid}" ]]; then
-    ${DOCKER_CMD} rm -f "${master_cid}"
+    ${DOCKER_CMD} rm -f "${master_cid}" > /dev/null
   fi
 
   local node_cids="$(${DOCKER_CMD} ps -qa --filter "name=${NODE_PREFIX}")"
   if [[ "${node_cids}" ]]; then
     node_cids=(${node_cids//\n/ })
     for cid in "${node_cids[@]}"; do
-      ${DOCKER_CMD} rm -f "${cid}"
+      ${DOCKER_CMD} rm -f "${cid}" > /dev/null
     done
   fi
 
-  echo "Cleanup up configuration to avoid conflict with a future cluster"
+  # Cleaning up configuration to avoid conflict with a future cluster
   # The container will have created configuration as root
-  sudo rm -rf ${CONFIG_ROOT}/openshift.local.*
+  sudo rm -rf ${config_root}/openshift.local.*
 
   # Cleanup orphaned volumes
   #
   # See: https://github.com/jpetazzo/dind#important-warning-about-disk-usage
   #
-  echo "Cleaning up volumes used by docker-in-docker daemons"
   local volume_ids=$(${DOCKER_CMD} volume ls -qf dangling=true)
   if [[ "${volume_ids}" ]]; then
-    ${DOCKER_CMD} volume rm ${volume_ids}
+    ${DOCKER_CMD} volume rm ${volume_ids} > /dev/null
   fi
 }
 
-# Build and deploy openshift binaries to an existing cluster
-function redeploy() {
-  local node_service="openshift-node"
+function check-selinux() {
+  if [[ "$(getenforce)" = "Enforcing" ]]; then
+    >&2 echo "Error: This script is not compatible with SELinux enforcing mode."
+    exit 1
+  fi
+}
 
-  ${DOCKER_CMD} exec -t "${MASTER_NAME}" bash -c "\
-. ${SCRIPT_ROOT}/provision-util.sh ; \
-os::provision::build-origin ${DEPLOYED_ROOT} ${SKIP_BUILD}"
+function get-network-plugin() {
+  local plugin=$1
 
-  echo "Stopping ${MASTER_NAME} service(s)"
-  ${DOCKER_CMD} exec -t "${MASTER_NAME}" systemctl stop "${MASTER_NAME}"
-  if [[ "${SDN_NODE}" = "true" ]]; then
-    ${DOCKER_CMD} exec -t "${MASTER_NAME}" systemctl stop "${node_service}"
-  fi
-  echo "Updating ${MASTER_NAME} binaries"
-  ${DOCKER_CMD} exec -t "${MASTER_NAME}" bash -c \
-". ${SCRIPT_ROOT}/provision-util.sh ; \
-os::provision::install-cmds ${DEPLOYED_ROOT}"
-  echo "Starting ${MASTER_NAME} service(s)"
-  ${DOCKER_CMD} exec -t "${MASTER_NAME}" systemctl start "${MASTER_NAME}"
-  if [[ "${SDN_NODE}" = "true" ]]; then
-    ${DOCKER_CMD} exec -t "${MASTER_NAME}" systemctl start "${node_service}"
+  local subnet_plugin="redhat/openshift-ovs-subnet"
+  local multitenant_plugin="redhat/openshift-ovs-multitenant"
+  local default_plugin="${subnet_plugin}"
+
+  if [[ "${plugin}" != "${subnet_plugin}" &&
+          "${plugin}" != "${multitenant_plugin}" &&
+          "${plugin}" != "cni" ]]; then
+    if [[ -n "${plugin}" ]]; then
+      >&2 echo "Invalid network plugin: ${plugin}"
```

Comment:
I feel like it's more often we put stream misdirection at the end of the statement, but that's a tiny nit